File: APITest/APITest/ViewController.m
//
//  ViewController.m
//  APITest
//
//  Created by alexiuce  on 2017/10/11.
//  Copyright © 2017年 alexiuce . All rights reserved.
//

#import "ViewController.h"
#import <AFNetworking.h>
#import "NSString+Extension.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    

    
    [self sendRequest:nil];
    

    
    
    // Do any additional setup after loading the view.
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}
- (IBAction)sendRequest:(id)sender {
    
    AFHTTPSessionManager *manager = [AFHTTPSessionManager manager];
    manager.requestSerializer = [AFHTTPRequestSerializer serializer];
//    NSString * voipServer = @"http://192.168.0.185:8080/common/saveDialVoIp";
    
    
    NSString *httpServer = @"http://httpbin.org/post";
    NSString *jsonText = @"{\"id\":\"0\",\"cust_id\":\"34\",\"link_id\":\"23\",\"number\":\"17682340610\",\"login_id\":\"1\",\"duration\":\"52\",\"call_time\":\"2017-10-12 09:59:04\",\"path\":\"record/17682340610/2017-10-12 09.59.04(17682340610).wav\",\"paid\":\"0.1\"}";
    NSString *jsonBase64 =  [[jsonText custom_Decrypt] base64String];
    //@"BVwXGlxETlJcHQsNCiEXGlxEXFxSXBIXEBUhFxpcRFxcUlwQCxMcGwxcRFxPSUhGTE1KTkhPTlxSXBIRGRcQIRcaXERcT1xSXBoLDB8KFxEQXERcT0xcUlwdHxISIQoXExtcRFxMTk9JU09OU09MXk9ORE5LRE1JXFJcDh8KFlxEXAwbHREMGlFPSUhGTE1KTkhPTlFMTk9JU09OU09MXk9OUE5LUE1JVk9JSEZMTUpOSE9OV1AJHwhcUlwOHxcaXERcTlBPXAM=";
   
    NSURL *url = [NSURL URLWithString:httpServer];
//
    NSMutableURLRequest *requestM = [NSMutableURLRequest requestWithURL:url];
    
// NSMutableURLRequest *requestM = [[AFHTTPRequestSerializer serializer] requestWithMethod:@"POST" URLString:httpServer parameters:nil error:nil];
    

    requestM.HTTPMethod = @"POST";
//    [requestM setValue:@"application/json" forHTTPHeaderField:@"Accept"];
    [requestM setValue:@"text/html;charset=utf-8" forHTTPHeaderField:@"Content-Type"];
    requestM.HTTPBody = [jsonBase64 dataUsingEncoding:NSUTF8StringEncoding];
    NSURLSessionDataTask *task =  [manager dataTaskWithRequest:requestM completionHandler:^(NSURLResponse * _Nonnull response, id  _Nullable responseObject, NSError * _Nullable error) {
        
        if (error) {
            NSLog(@"error : %@",error.localizedDescription);
            return ;
        }
        NSLog(@"response:  %@",responseObject);
        
    }];
    
    [task resume];

    
    
}


@end

File: TreeDemo/TreeDemo/ViewController.swift
//
//  ViewController.swift
//  TreeDemo
//
//  Created by alexiuce  on 2017/9/4.
//  Copyright © 2017年 alexiuce . All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    @IBOutlet weak var outline: NSOutlineView!
    
    
   lazy var nodes : [Node] = {
        let n1 = Node("One")
        let n11 = Node("One-1")
        let n12 = Node("One-2")
        n1.children = [n11,n12]
        
        let n2 = Node("Two")
        let n21 = Node("Two-1")
        n2.children = [n21]
        
        let n3 = Node("Three")
        
        return [n1, n2,n3]
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
    
       
        // Do any additional setup after loading the view.
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
}

// MARK: NSOutlineView Delegate Mothed

extension ViewController : NSOutlineViewDelegate{
    func outlineView(_ outlineView: NSOutlineView, shouldSelectItem item: Any) -> Bool {
        // 1. item 为将要选择的数据模型对象
        print("select item \(item.self)")
        // 2. 这里可以处理用户选择时的业务逻辑，比如根据选中是数据模型进行相应的网络请求，
        return true
    }
}


